### DISCUSSION TITLE
Regex pattern at sentence level (not at token level)

### DISCUSSION THREAD

**Discussion by Javier-Jimenez99** [explosion/spaCy#8797] [Help: Coding & Implementations] [answered]
There should be a way to extract entities at sentence level using regex. 

With `EntityRuler` can be extracted entities but at token level and with `PhraseMatcher` cant be used regex.

How can I do it?

**Comment by polm** [comment, marked as answer]
You can just use code instead of using a Matcher, and use [Doc.char_span](https://spacy.io/api/doc#char_span) to label entities. 

```
ents = []
doc = nlp(text)
for sent in doc.sents:
    match = re.find(...): # find a match
    if not match: continue
    ent = sent.char_span(match.start, match.end, label="SOMETHING")
    ents.append(ent)
doc.ents = ents
```

The Matchers are designed to give a good balance of speed, convenience, and versatility, but they don't cover every situation.
